src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,71 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int ARRSIZE = 10;
+    private static final int CAPACITY = 3 / 2;
+    private int size;
+    private T[] array;
 
+    public MateStack() {
+        array = (T[]) new Object[ARRSIZE];
+    }
+
+    public void push(T value) {
+        if (size == array.length) {
+            array = grow();
+        }
+        array[size++] = value;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return elementAt(size - 1);
     }
 
     public T pop() {
-        return null;
+        T obj;
+        obj = peek();
+        removeElementAt(size - 1);
+        return obj;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private int newCapacity() {
+        return size * CAPACITY;
+    }
+
+    private T[] grow() {
+        T[] growArray = (T[]) new Object[newCapacity()];
```

Comment:
```suggestion
        T[] grownArray /resizedArray = (T[]) new Object[newCapacity()];
```
better to use nouns for variable names